timeseries/src/autogluon/timeseries/models/presets.py

```diff
@@ -151,18 +156,23 @@ def get_preset_models(
     will create models according to presets.
     """
     models = []
-    if isinstance(hyperparameters, str):
+    if hyperparameters is None:
+        hp_string = "default_hpo" if hyperparameter_tune else "default"
+        hyperparameters = copy.deepcopy(get_default_hps(hp_string, prediction_length))
+    elif isinstance(hyperparameters, str):
         hyperparameters = copy.deepcopy(get_default_hps(hyperparameters, prediction_length))
+    elif isinstance(hyperparameters, dict):
+        default_hps = copy.deepcopy(get_default_hps("default", prediction_length))
+        updated_hyperparameters = {}
+        for model, hps in hyperparameters.items():
+            updated_hyperparameters[model] = default_hps.get(model, {})
+            updated_hyperparameters[model].update(hps)
+        hyperparameters = copy.deepcopy(updated_hyperparameters)
     else:
-        hp_str = "default" if not hyperparameter_tune else "default_hpo"
-        default_hps = copy.deepcopy(get_default_hps(hp_str, prediction_length))
-
-        if hyperparameters is not None:
-            # filter only default_hps for models with hyperparameters provided
-            default_hps = {model: default_hps.get(model, {}) for model in hyperparameters}
-            for model in hyperparameters:
-                default_hps[model].update(hyperparameters[model])
-        hyperparameters = copy.deepcopy(default_hps)
+        raise ValueError(
+            f"hyperparameters must be a dict, a string or None (received {type(hyperparameters)}). "
+            "Please see the documentation for TimeSeriesPredictor.fit"
+        )
```

Comment:
Updated logic for hyperparameter handling (that fixes a bug)
- Now: If `hyperparameters` is a dict, we take the default presets from `default`, and override them based on `hyperparameters`.
- Before: `hyperparameters` is a dict and `hyperparameter_tune == True`, we take the default presets from `default_hpo`, and override them based on `hyperparameters`. This behavior resulted in the HPO search space potentially including parameters not provided by the user.